IT2901-24-2018/vapi
# Story25 refactor

## Types of changes


## How to review
- Check if the tests in test_segmenting.py test what they should
  - Do they test every element they should?
  - Do they check things that will always be true? 
- Check if I've implemented the segmenter correctly so that what views.py sends is actually correct
  - Do the new segments actually end up in the database?
- Check on the segments and see if they are actually segmented correctly 
  - Are there more segments than there originally were?
  - Are there any missing coordinates points?
  - Are there duplicate segments?
  - Are the attributes on each new segment correct?

## Description

This refactors the segmenting so that it happens in views.py instead of outside. It also changes the test files so  that they can be run on the new segmentation. It fixes some imports here and there, and moves files from. Oh also it adds a couple tests. It now checks the filter_road function, checks for missing coordinates, and checks if the coordinates lists are the correct type at different stages of segmentation. 

## Motivation and Context


It didn't make sense to do segmenting where we did before. It makes more sense to segment the road network as we pull it in. That's why it's now in the views.py. 
Resolves #138

## How Has This Been Tested?



I ran the tests and ran the example_roadnet_to_db.py and it looked ok but I'm an idiot what do I know

## Screenshots (if appropriate):

## Checklist:
